apps/wallet/src/background/Alarms.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import Browser from 'webextension-polyfill';
+
+import Keyring from './Keyring';
+
+export const LOCK_ALARM_NAME = 'lock-keyring-alarm';
+
+class Alarms {
+    public async setLockAlarm() {
+        if (Keyring.isLocked) {
+            return;
+        }
+        const delayInMinutes = 5;
```

Comment:
This really should be a parameter to allow this to be configured.

Reply:
uses a user setting in a following PR